// vk-api/index.js
var httpsRequest = require("https-request");
var querystring = require("querystring");
var vkConstants = require("./vk-constants");

var token = "8ae373703fb582931b25ebf80d4846a32292d22a3000c8a479d57883094568cccd9aa82246db7e82eec66";

function constructRequestUrl (url, methodName, params) {
	return url + methodName + "?" + querystring.stringify(params);
} 

function requestVkApi (methodName, params, cb) {
	var url = constructRequestUrl(vkConstants.API_REQUEST_URL, methodName, params);
	
	httpsRequest.get(url, function (response) {
		cb(response);
	});
}

module.exports = function VkApiConstructor () {
	this.postOnWall = function (content, cb) {
		var ownerId = content.ownerId;
		var message = content.text;
		var methodName = "wall.post";
		var parameters = {
		    	owner_id: ownerId,
		    	from_group: 1,
		    	message: message,
		    	//message: encodeURIComponent(message),
		    	access_token: token
			};

		requestVkApi(methodName, parameters, cb);
	};

	this.getFriends = function (cb) {
		var methodName = "friends.get";
		var parameters = {
		    user_id: vkConstants.MY_VK_ID,
		    fields: vkConstants.FRIENDS_FIELDS.join(","),
		    access_token: token
		};

		requestVkApi(methodName, parameters, cb);
	};

	this.getGroups = function (cb) {
		var methodName = "groups.get";
		var parameters = {
		    user_id: vkConstants.MY_VK_ID,
		    extended: 1,
		    filter: "admin",
		    fields: vkConstants.GROUPRS_FIELDS.join(","),
		    access_token: token
		};

		requestVkApi(methodName, parameters, cb);
	};

	this.requestAccessToken = function (cb) {
		var redirectPage = "https://oauth.vk.com/blank.html";
		var authTokenHostName = "https://oauth.vk.com/authorize?";
		var authParameters = {
			client_id: vkConstants.APP_ID,
			scope: requestedRights,
			redirect_uri: vkConstants.ACCESS_TYPES.join(","),
			display: "page",
			v: 5.45,
			response_type: "token"
		};

		var authURL = constructRequestUrl(authTokenHostName, "", authParameters);

		httpsRequest.get(authURL, function (r) {
			cb(r);
		});
	};
};
