// vk-api/vk-constants.js
module.exports = {
	APP_ID: 5337633,
	MY_VK_ID: 6718249,
	API_REQUEST_URL: "https://api.vkontakte.ru/method/",
	FRIENDS_FIELDS: [   
        "nickname",         "domain",                       "sex",                  "bdate",
        "city",             "country",                      "timezone",             "photo_50",
        "photo_100",        "photo_200_orig",               "has_mobile",           "contacts",
        "education",        "online",                       "relation",             "last_seen",
        "status",           "can_write_private_message",    "can_see_all_posts",    "can_post",
        "universities"
    ],
    GROUPRS_FIELDS: [
		"city", 			"country", 			"place", 				"description",
		"wiki_page", 		"members_count", 	"counters", 			"start_date",
		"finish_date", 		"can_post", 		"can_see_all_posts", 	"activity",
		"status", 			"contacts", 		"links", 				"fixed_post",
		"verified", 		"site", 			"can_create_topic"
	],
	ACCESS_TYPES: [
		"notify", 	"friends", 	"photos", 		"audio", 
		"video", 	"docs", 	"notes", 		"pages", 
		"status", 	"offers", 	"questions", 	"wall", 
		"groups", 	"email", 	"notifications","stats", 	
		"ads", 		"market", 	"offline"
	],
	methods: {
		GET_FRIENDS: "friends.get",
		GET_GROUPS: "groups.get"
	}

};
